src/rules/javascript/deprecated_entities.js

```diff
@@ -0,0 +1,42 @@
+import { NO_DOCUMENT_WRITE, NSI_DNS_SERVICE_RESOLVE,
+         NSI_SOUND_PLAY } from 'messages';
+import { getNodeReferenceName } from 'utils';
+
+export const DEPRECATED_ENTITIES = [
+  {
+    error: NO_DOCUMENT_WRITE,
+    object: 'document',
+    property: 'write',
+  }, {
+    error: NSI_DNS_SERVICE_RESOLVE,
+    object: 'nsIDNSService',
+    property: 'resolve',
+  }, {
+    error: NSI_SOUND_PLAY,
+    object: 'nsISound',
+    property: 'play',
+  },
+];
+
+
+export function deprecated_entities(context) {
+  return {
+    CallExpression: function(node) {
+      // We're only looking for calls that look like `foo.bar()`.
+      if (typeof node.callee.object !== 'undefined' &&
```

Comment:
I assume `node.callee` is always defined? (I forget, AST spec is hard to remember ^_^)